### PR TITLE
hexdiff: init at unstable-2018-01-24

### PR DESCRIPTION
**Pull request by rogarb**
###### Description of changes

A command line tool to display differences in binary files using a hexadecimal formatting.
[https://github.com/ahroach/hexdiff]()

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/hexdiff/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "hexdiff";
+  version = "unstable-2018-01-24";
+  src = fetchFromGitHub {
```
Comment (AndersonTorres) [inline pkgs/tools/misc/hexdiff/default.nix:6]:
```suggestion

  src = fetchFromGitHub {
```

File: pkgs/tools/misc/hexdiff/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    owner = "ahroach";
+    repo = "${pname}";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/hexdiff/default.nix:8]: Don't parameterize `pname`.

```suggestion
    repo = "hexdiff";
```

File: pkgs/tools/misc/hexdiff/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    rev = "3e96f27e65167c619ede35ab04232163dc273e69";
+    sha256 = "sha256-G6Qi7e4o+0ahcslJ8UfJrdoc8NNkY+nl6kyDlkJCo9I=";
+  };
+
+  buildPhase = "$CC -o ${pname} ${pname}.c";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/hexdiff/default.nix:13]:
```suggestion
  dontConfigure = true;

  buildPhase = ''
    runHook preBuild
    
    ${stdenv.cc.targetPrefix}cc -o hexdiff hexdiff.c
    
    runHook postBuild
  '';
```

File: pkgs/tools/misc/hexdiff/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ${pname} $out/bin/${pname}
+  '';
```
Comment (AndersonTorres) [inline pkgs/tools/misc/hexdiff/default.nix:17]:
```suggestion

  installPhase = ''
    runHook preInstall
    
    install -D hexdiff  -t $out/bin/
    
    runHook postInstall
  '';
```

File: pkgs/tools/misc/hexdiff/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  meta =  with lib; {
+    description = "A terminal application for differencing two binary files, with color-coded output";
+    license = licenses.gpl3Plus;
+    platforms = platforms.linux;
+    maintainers = with lib.maintainers; [ rogarb ];
+    homepage = "https://github.com/ahroach/hexdiff";
+  };
```
Comment (AndersonTorres) [inline pkgs/tools/misc/hexdiff/default.nix:25]:
```suggestion
  meta = {
    homepage = "https://github.com/ahroach/hexdiff";
    description = "A terminal application for differencing two binary files, with color-coded output";
    license = lib.licenses.gpl3Plus;
    maintainers = with lib.maintainers; [ rogarb ];
    platforms = lib.platforms.linux;
  };
```

File: pkgs/tools/misc/hexdiff/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/tools/misc/hexdiff/default.nix:3]:
```suggestion
stdenv.mkDerivation {
```

Avoid using `rec`, especially when not needed.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by rogarb** [comment]
@AndersonTorres thank you for your input. I applied your suggestions.

**Comment by AndersonTorres** [comment]
Use rebase instead of merge.

**Comment by rogarb** [comment]
> Use rebase instead of merge.

It seems I messed up with the rebase. I will try to fix it tomorrow.

**Pull request closed without merging**